sympy/core/expr.py

```diff
@@ -576,6 +595,8 @@ def is_constant(self, *wrt, **flags):
                         # dead line provided _random returns None in such cases
                         return None
                 return False
+        if check_denominator_zeros(self) or check_denominator_zeros(1/self):
```

Comment:
This function is much better for me. But we have to check zero expression and 1/expression:

`In [1]: from sympy import *

In [2]: x = Symbol('x', real=True)

In [3]: pp = tan(x)*cot(x)

In [4]: pp.is_constant()
Out[4]: False`
with 1/expression and 
`In [11]: pp = tan(x)*cot(x)

In [12]: pp.is_constant()
Out[12]: True
`
without.